HLTrigger/JetMET/plugins/HLTL1TMatchedJetsVBFFilter.h

```diff
@@ -0,0 +1,326 @@
+      maxJetDeltaR2_(maxJetDeltaR_ * maxJetDeltaR_),
+      l1tJetRefsToken_(matchJetsByDeltaR_ ? consumes<trigger::TriggerFilterObjectWithRefs>(
+                                                iConfig.getParameter<edm::InputTag>("l1tJetRefs"))
+                                          : edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs>()),
+      algorithm_(getAlgorithmFromString(iConfig.getParameter<std::string>("algorithm"))),
+      minPt1_(iConfig.getParameter<double>("minPt1")),
+      minPt2_(iConfig.getParameter<double>("minPt2")),
+      minPt3_(iConfig.getParameter<double>("minPt3")),
+      minInvMass_(iConfig.getParameter<double>("minInvMass")),
+      minNJets_(iConfig.getParameter<int>("minNJets")),
+      maxNJets_(iConfig.getParameter<int>("maxNJets")),
+      triggerType_(iConfig.getParameter<int>("triggerType")) {
+  // delta-R matching
+  if (matchJetsByDeltaR_ and maxJetDeltaR_ < 0) {
+    throw cms::Exception("InvalidConfigurationParameter")
+        << "invalid value for parameter \"maxJetDeltaR\" (must be > 0): " << maxJetDeltaR_;
+  }
+
+  // values of minPt{1,2,3} thresholds
+  if (minPt2_ > minPt1_) {
+    throw cms::Exception("InvalidConfigurationParameter")
+        << "minPt1 (" << minPt1_ << ") must not be smaller than minPt2 (" << minPt2_ << ")";
+  } else if (minPt3_ > minPt1_) {
+    throw cms::Exception("InvalidConfigurationParameter")
+        << "minPt1 (" << minPt1_ << ") must not be smaller than minPt3 (" << minPt3_ << ")";
+  } else if (minPt2_ > minPt3_) {
+    throw cms::Exception("InvalidConfigurationParameter")
+        << "minPt3 (" << minPt3_ << ") must not be smaller than minPt2 (" << minPt2_ << ")";
+  }
+}
+
+template <typename T>
+typename HLTL1TMatchedJetsVBFFilter<T>::Algorithm HLTL1TMatchedJetsVBFFilter<T>::getAlgorithmFromString(
+    std::string const& str) const {
+  if (str == "VBF") {
+    return Algorithm::VBF;
+  } else if (str == "VBFPlus2CentralJets") {
+    return Algorithm::VBFPlus2CentralJets;
+  }
+
+  throw cms::Exception("HLTL1TMatchedJetsVBFFilterInvalidAlgorithmName")
+      << "invalid value for argument of getAlgorithmFromString method: " << str
+      << " (valid values are \"VBF\" and \"VBFPlus2CentralJets\")";
+}
+
+template <typename T>
+void HLTL1TMatchedJetsVBFFilter<T>::fillJetIndices(std::vector<unsigned int>& outputJetIndices,
+                                                   std::vector<T> const& jets,
+                                                   std::vector<unsigned int> const& jetIndices,
+                                                   double const pt1,
+                                                   double const pt3,
+                                                   double const mjj) const {
+  // reset output jet indices
+  outputJetIndices.clear();
+
+  // find dijet pair with highest Mjj
+  int i1 = -1;
+  int i2 = -1;
+  double m2jj_max = -1;
+
+  for (unsigned int i = 0; i < jetIndices.size() - 1; i++) {
```

Comment:
The use of `size() - 1` here is a bug, missed during the integration of this PR, spotted in https://github.com/cms-sw/cmssw/pull/43300#issuecomment-1817465701, and fixed in #43324 (14_0_X), #43325 (13_3_X) and #43326 (13_2_X)